#! /bin/sh /usr/share/dpatch/dpatch-run
## 103_fix-crash-in-i830_emit_state.dpatch by Chris Coulson <chrisccoulson@googlemail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: This patch fixes a crash in i830_emit_state. 
## DP: Upstream commit - 48c29b60a899877c76d643b9cd06d5277cd97b9c
## DP: Upstream bug report - https://bugs.freedesktop.org/show_bug.cgi?id=17766
## DP: Ubuntu bug report - https://bugs.launchpad.net/ubuntu/+source/mesa/+bug/277709

@DPATCH@
diff -urNad mesa-7.2~/src/mesa/drivers/dri/i915/i830_vtbl.c mesa-7.2/src/mesa/drivers/dri/i915/i830_vtbl.c
--- mesa-7.2~/src/mesa/drivers/dri/i915/i830_vtbl.c	2008-08-25 15:49:40.000000000 +0100
+++ mesa-7.2/src/mesa/drivers/dri/i915/i830_vtbl.c	2008-10-21 17:43:44.000000000 +0100
@@ -444,7 +444,8 @@
    ret = 0;
    if (dirty & I830_UPLOAD_BUFFERS) {
      ret |= dri_bufmgr_check_aperture_space(state->draw_region->buffer);
-     ret |= dri_bufmgr_check_aperture_space(state->depth_region->buffer);
+     if (state->depth_region)
+     	ret |= dri_bufmgr_check_aperture_space(state->depth_region->buffer);
    }
    
    for (i = 0; i < I830_TEX_UNITS; i++)
